lib/light-service/context/key_verifier.rb

```diff
@@ -40,13 +40,13 @@ def verify
       context
     end
 
-    def self.verify_keys(context, &block)
-      ReservedKeysVerifier.new(context).verify
-      ExpectedKeyVerifier.new(context).verify
+    def self.verify_keys(context, action, &block)
+      ReservedKeysVerifier.new(context, action).verify
+      ExpectedKeyVerifier.new(context, action).verify
 
       block.call
 
-      PromisedKeyVerifier.new(context).verify
+      PromisedKeyVerifier.new(context, action).verify
     end
```

Comment:
@ramontayag: I ran all the specs and they pass in your PR. However, I am trying to understand why we want to do this before I merge the code in. Here is what puzzles me.

I added this to line 44:
```ruby
pp context.current_action == action
```

This evaluates to true. Then I changed line 45 to `ReservedKeysVerifier.new(context, context.current_action).verify`. I changed the same for ExpectedKeyVerifier and PromisedKeyVerifier initializer. When I run the tests, I am getting the PromisedKeyNotFoundError.

Do you know why? Can you explain?